packages/app-config-writer/src/smartlinks-config/utils.ts

```diff
@@ -0,0 +1,202 @@
+import type { AxiosBasicCredentials } from 'axios';
+import { cyan } from 'chalk';
+import { render } from 'ejs';
+import type { Editor } from 'mem-fs-editor';
+import { join } from 'path';
+import { create } from '@sap-ux/axios-extension';
+import { getDestinationUrlForAppStudio, isAppStudio } from '@sap-ux/btp-utils';
+import type { ToolsLogger } from '@sap-ux/logger';
+import { FileName } from '@sap-ux/project-access';
+import { BackendSystemKey, getService } from '@sap-ux/store';
+import type { BackendSystem } from '@sap-ux/store';
+import { t } from '../i18n';
+import { getTemplatePath } from '../templates';
+import { UrlParameters } from '../types';
+import type {
+    InboundTargetsConfig,
+    SmartLinksSandboxConfig,
+    SystemDetailsResponse,
+    TargetConfig,
+    TargetMapping
+} from '../types';
+import { addUi5YamlServeStaticMiddleware, readUi5DeployConfigTarget } from './ui5-yaml';
+
+/**
+ * @description Check the secure storage if it has credentials for the entered url.
+ * @param url target system url
+ * @param client optional sap-client parameter
+ * @param logger Logger for user output
+ * @returns credentials or undefined
+ */
+export async function getLocalStoredCredentials(
+    url: string,
+    client?: string,
+    logger?: ToolsLogger
+): Promise<AxiosBasicCredentials | undefined> {
+    // check if system credentials are stored in the store
+    try {
+        const systemStore = await getService<BackendSystem, BackendSystemKey>({ logger, entityName: 'system' });
+        const system = await systemStore.read(new BackendSystemKey({ url, client }));
+        return system?.username ? { username: system.username, password: system.password || '' } : undefined;
+    } catch (error) {
+        logger?.warn(t('warnings.useCredentialsFailed'));
+        logger?.debug(error as object);
+    }
+    return undefined;
+}
+
+/**
+ * @description Combines the target parameters and returns the target url
+ * @param baseUrl destination or url
+ * @param client optional sap-client
+ * @returns target url to be called for smartlinks config
+ */
+function getSmartLinksTargetUrl(baseUrl: string, client?: string) {
```

Comment:
That is not exactly how the `axios-extension` is supposed to be used. You shouldn't build full urls here. I will create a PR with a different way and link it here.

Reply:
Here we go: https://github.com/SAP/open-ux-tools/pull/953. I haven't tested it but this should be doing the same as your previous code just leaving the url generation to the `axios-extension`.
(and adding support for url usage in BAS ;) )

Reply:
Hi @tobiasqueck thanks for the axios-extension suggestion. Locally this works great, but on BAS I run into 404 providing either url or destination. Is there some approach I am missing?

Reply:
Please have a look at https://github.com/SAP/open-ux-tools/pull/913/commits/cf42b02c84071400ad8c55d3ca7f8d95f09291be. It should fix it. The actual issue is in the `axios-extension` but that needs to be addressed separately, so here is a local fix for you.